src/copy.c

```diff
@@ -396,14 +396,41 @@ copyfrom(CopyChunkState *ccstate, List *range_table, Hypertable *ht, void (*call
 				resultRelInfo->ri_RelationDesc->rd_att->constr)
 			{
 				Assert(resultRelInfo->ri_RangeTableIndex > 0 && estate->es_range_table);
-				ExecConstraints(resultRelInfo, myslot, estate);
+				ExecConstraints(saved_resultRelInfo, myslot, estate);
```

Comment:
why is this on the saved_resultRelInfo (this would be the hypertable, isn't it?). Shouldn't this be on the chunk itself? We need to special case only for the compressed chunk case.